packages/zipkin-instrumentation-request-promise/test/integrationTest.js

```diff
@@ -262,7 +275,91 @@ describe('request instrumentation - integration test', () => {
           expect(annotations[5].annotation.annotationType).to.equal('BinaryAnnotation');
           expect(annotations[5].annotation.key).to.equal('error');
           expect(annotations[5].annotation.value)
-            .to.contain('Error: getaddrinfo ENOTFOUND bad.invalid.url bad.invalid.url:80');
+            .to.contain('Error: connect ECONNREFUSED 127.0.0.1:12345');
+
+          expect(annotations[6].annotation.annotationType).to.equal('ClientRecv');
+
+          expect(annotations[7]).to.be.undefined; // eslint-disable-line no-unused-expressions
+          done();
+        });
+      });
+    });
+  });
+
+  it('should report when service returns 400', done => {
+    tracer.scoped(() => {
+      const apiServer = api.listen(0, () => {
+        const apiPort = apiServer.address().port;
+        const apiHost = '127.0.0.1';
+        const zipkinRequest = new Request(tracer, remoteServiceName);
+        const urlPath = '/weather/securedTown';
+        const url = `http://${apiHost}:${apiPort}${urlPath}`;
+        zipkinRequest.get({url, timeout: 100}).catch(() => {
+          const annotations = record.args.map(args => args[0]);
+          const initialTraceId = annotations[0].traceId.traceId;
+          annotations.forEach(ann => expect(ann.traceId.traceId)
+            .to.equal(initialTraceId).and
+            .to.have.lengthOf(16));
+
+          expect(annotations[0].annotation.annotationType).to.equal('ServiceName');
+          expect(annotations[0].annotation.serviceName).to.equal('weather-app');
+
+          expect(annotations[1].annotation.annotationType).to.equal('Rpc');
+          expect(annotations[1].annotation.name).to.equal('GET');
+
+          expect(annotations[2].annotation.annotationType).to.equal('BinaryAnnotation');
+          expect(annotations[2].annotation.key).to.equal('http.path');
+          expect(annotations[2].annotation.value).to.equal(urlPath);
+
+          expect(annotations[3].annotation.annotationType).to.equal('ClientSend');
+
+          expect(annotations[4].annotation.annotationType).to.equal('ServerAddr');
+
+          expect(annotations[5].annotation.annotationType).to.equal('BinaryAnnotation');
+          expect(annotations[5].annotation.key).to.equal('http.status_code');
+          expect(annotations[5].annotation.value).to.equal('400');
+
+          expect(annotations[6].annotation.annotationType).to.equal('ClientRecv');
+
+          expect(annotations[7]).to.be.undefined; // eslint-disable-line no-unused-expressions
+          done();
+        });
+      });
+    });
+  });
+
+  it('should report when service returns 500', done => {
```

Comment:
@mkante I believe we should add an error tag on status codes 4xx and 5xx. We do so in other http client libraries: https://github.com/openzipkin/zipkin-go/blob/master/middleware/http/client.go#L119 and https://github.com/jcchavezs/zipkin-instrumentation-guzzle/blob/master/src/ZipkinGuzzle/Middleware.php#L61

I am surprised we don't do this in https://github.com/openzipkin/zipkin-js/blob/master/packages/zipkin-instrumentation-request/test/integrationTest.js#L231.

Any input @adriancole.

Reply:
Hm I see want you mean

Reply:
Sounds good to me. Any opinion on this @adriancole?

Reply:
usually also check <200 (though this should be not usual :P)

Reply:
@mkante can you please add that change as well?

Reply:
On this topic I just opened this PR as an attempt to uniformize this errors: https://github.com/openzipkin/zipkin-js/pull/261 cc @mkante @adriancole

Reply:
Thanks I will look into this weekend